src/util.rs

```diff
@@ -176,14 +189,35 @@ pub mod tests {
     }
 
     #[test]
-    fn test_unpack_share() {
+    fn test_unpack_share_mut() {
         let dim = 15;
         let len = proof_length(dim);
 
         let mut share = vec![Field32::from(0); len];
         let unpacked = unpack_proof_mut(&mut share, dim).unwrap();
         *unpacked.f0 = Field32::from(12);
         assert_eq!(share[dim], 12);
+
+        let mut short_share = vec![Field32::from(0); len - 1];
+        assert_matches!(
+            unpack_proof_mut(&mut short_share, dim),
+            Err(ProofError::InputSizeMismatch)
+        );
+    }
+
+    #[test]
+    fn test_unpack_share() {
+        let dim = 15;
+        let len = proof_length(dim);
+
+        let share = vec![Field64::from(0); len];
+        unpack_proof(&share, dim).unwrap();
+
+        let short_share = vec![Field64::from(0); len - 1];
+        assert_matches!(
+            unpack_proof(&short_share, dim),
+            Err(ProofError::InputSizeMismatch)
+        );
```

Comment:
Could you write this assertion without `assert_matches!()`? Something like
```rust
assert_eq!(unpack_proof(...), Err(...));
```

Reply:
My compiler won't let me: it complains that:
```
error[E0369]: binary operation `==` cannot be applied to type `std::result::Result<UnpackedProofMut<'_, field::Field32>, util::ProofError>`
   --> src/util.rs:202:9
    |
202 | /         assert_eq!(
203 | |             unpack_proof_mut(&mut short_share, dim),
204 | |             Err(ProofError::InputSizeMismatch)
205 | |         );
    | |          ^
    | |          |
    | |__________std::result::Result<UnpackedProofMut<'_, field::Field32>, util::ProofError>
    |            std::result::Result<_, util::ProofError>
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

We could do:

```
    let err = unpack_proof_mut(&mut short_share, dim).unwrap_err();
    assert_eq!(err, ProofError::InputSizeMismatch);
```
... but for that to work I'd have to provide a `PartialEq` impl on `ProofError` and I think it's easier to use `assert_matches!`.

Reply:
Works for me.